frontend/public/graphql/client.ts

```diff
@@ -50,5 +50,5 @@ export const fetchK8s = <R extends K8sResourceCommon = K8sResourceCommon>(
   kind: K8sKind,
   name?: string,
   ns?: string,
-  opts?: { [k: string]: string },
+  opts?: { [k: string]: string } | { queryParams: { [k: string]: string } },
```

Comment:
lets make it even more specifc
```suggestion
  opts?: { 
    path?: string;
    queryParams?: {
      [k: string]: string;
    };
  },
```

Reply:
ack

Reply:
Shouldn't we need to add `[k: string]: string` too here for handling other options like `ns?: string, name?: string`

Reply:
I dont like duplication.

what about 

```
export const fetchK8s = <R extends K8sResourceCommon = K8sResourceCommon>(
  kind: K8sKind,
  name?: string,
  ns?: string,
  path?: string,
  queryParams: { [key: string]: string },
```

would that work for you ?

Reply:
Yes, this will work for all the options that are acceptable inside `getK8sResourcePath`.